### ISSUE TITLE
GenericOneshotBroadcastReceiver incorrectly requires MutexType to be Clone

### ISSUE THREAD

**Issue by CSCPF2** [Matthias247/futures-intrusive#35]
The `#[derive(Clone)]` macro thinks `MutexType` needs to be `Clone` as well in this code:
https://github.com/Matthias247/futures-intrusive/blob/4592e5e837d64dfa8f5138acb6c253e8868f5aee/src/channel/oneshot_broadcast.rs#L315-L328

This isn't intended, is it?

If not, I'd suggest implementing a hand-rolled `Clone` instead.

**Comment by Matthias247** [comment]
I am not sure why this would be derived wrong? The type only contains an `Arc`,  and `Arc` should be `Clone`:

```rust
impl<T> Clone for Arc<T> where
    T: ?Sized, 
```

**Comment by CSCPF2** [comment]
@Matthias247 You can see the problem in [the docs](https://docs.rs/futures-intrusive/0.3.0/futures_intrusive/channel/shared/struct.GenericOneshotBroadcastReceiver.html#implementations). The impl for `Clone` is
```rust
impl<MutexType: Clone, T: Clone> Clone for GenericOneshotBroadcastReceiver<MutexType, T> where
    MutexType: RawMutex,
    T: Clone + 'static,
```
whereas you want
```rust
impl<MutexType, T: Clone> Clone for GenericOneshotBroadcastReceiver<MutexType, T> where
    MutexType: RawMutex,
    T: Clone + 'static, // (this is still redundant but is besides the point)
```
.

This happens because the generic arguments are `<MutexType, T>`. `#[derive(Clone)]` thinks both need to be `Clone`.

You'll just have to impl `Clone` by hand.

**Comment by CSCPF2** [comment]
Hope you can make a release after fixing this. The `channel::shared::oneshot_broadcast_channel` API is unusable in its current form.

**Comment by Matthias247** [comment]
Fixed in 0.3.1
